src/main/java/com/incentives/piggyback/user/controller/UserController.java

```diff
@@ -52,6 +67,16 @@
     public ResponseEntity<User> updateUser(@PathVariable Long id, @Valid @RequestBody User user) {
         log.debug("User Service: Received PUT request for updating user with userid."+ id);
         if(userService.findById(id).isPresent()) {
+            User updatedUser = userService.save(user);
+            //PUSHING MESSAGES TO GCP
+            messagingGateway.sendToPubsub(
+                    CommonUtility.stringifyEventForPublish(
+                            updatedUser.toString(),
+                            Constant.USER_UPDATED_EVENT,
+                            Calendar.getInstance().getTime().toString(),
+                            HttpStatus.OK.toString(),
```

Comment:
4th event field is meant for specifying partnerId, so that an event can be mapped to a partner - for billing and invoicing. For events not related to partners directly, it should be passed as empty.